src/libs/sfdk/dockervirtualmachine.cpp

```diff
@@ -0,0 +1,529 @@
+** General Public License version 2.1 as published by the Free Software
+** Foundation and appearing in the file LICENSE.LGPL included in the
+** packaging of this file.  Please review the following information to
+** ensure the GNU Lesser General Public License version 2.1 requirements
+** will be met: http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
+**
+** Other Usage
+**
+** Alternatively, this file may be used in accordance with the terms and
+** conditions contained in a signed written agreement between you and
+** The Qt Company.
+**
+****************************************************************************/
+
+#include "dockervirtualmachine_p.h"
+
+#include "asynchronous_p.h"
+#include "sfdkconstants.h"
+#include "sdk_p.h"
+#include "usersettings_p.h"
+
+#include <utils/fileutils.h>
+#include <utils/optional.h>
+#include <utils/qtcassert.h>
+
+#include <QDir>
+#include <QGlobalStatic>
+#include <QJsonDocument>
+#include <QJsonObject>
+#include <QProcess>
+#include <QRegularExpression>
+
+namespace {
+const char DOCKER[] = "docker";
+const char RUN[] = "run";
+const char PRIVILEGED1[] = "--privileged";
+const char PRIVILEGED2[] = "--cap-drop=NET_ADMIN";
+const char STOP[] = "stop";
+const char COMMIT[] = "commit";
+const char CONTAINER_LIST[] = "ps";
+const char DETACHED[] = "-d";
+const char PUBLISH_PORT[] = "-p";
+const char VOLUME[] = "-v";
+const char CGROUP_MOUNT[] = "/sys/fs/cgroup:/sys/fs/cgroup:ro";
+const char TMPFS[] = "--tmpfs";
+const char TMPFS1[] = "/run:exec,mode=755";
+const char TMPFS2[] = "/run/lock";
+const char TMPFS3[] = "/tmp:exec";
+const char FILTER_PREFIX[] = "--filter=ancestor=";
+const char FORMAT_CONTAINER_LIST[] = "--format={{.Image}} {{.ID}}";
+const char COMMAND_INSPECT[] = "inspect";
+const char FORMAT_LABEL_LIST[] = "--format={{json .Config.Labels}}";
+const char IMAGE[] = "image";
+const char LIST[] = "ls";
+const char FORMAT_REPOSITORY[] = "--format={{.Repository}}";
+const char UNNAMED_IMAGE[] = "<none>";
+const char COMMAND_BUILD[] = "build";
+const char DOCKERFILE_PREFIX[] = "FROM ";
+const char LABEL_PREFIX[] = "--label=";
+const char TAG[] = "--tag";
+const quint16 GUESTSSHPORT = 22;
+const quint16 GUESTWWWPORT = 9292;
+}
+
+namespace Sfdk {
+
+void DockerVirtualMachinePrivate::fetchInfo(VirtualMachineInfo::ExtraInfos extraInfo,
+        const QObject *context, const Functor<const VirtualMachineInfo &, bool> &functor) const
+{
+    Q_Q(const DockerVirtualMachine);
+    Q_ASSERT(context);
+    Q_ASSERT(functor);
+
+    // TODO: fetch snapshot info if requested
+    Q_UNUSED(extraInfo)
+
+    QStringList arguments;
+    arguments.append(COMMAND_INSPECT);
+    arguments.append(q->name());
+    arguments.append(FORMAT_LABEL_LIST);
+
+    auto runner = std::make_unique<ProcessRunner>(dockerPath(), arguments);
+    QObject::connect(runner.get(), &ProcessRunner::done, q,
+                     [this, functor, process = runner->process()](bool ok) {
```

Comment:
`context` should be passed instead of `q`.

BTW, aligning arguments under opening brace often hits space limitations, so it can hardly be applied consistently, so something I would rather avoid. Please consider realigning such code to just plain two levels shift.

Reply:
Fixed. And modified my emacs indentation rules according to your suggestion.

Reply:
Nice :) There is one more case on line 155 left.

Reply:
fixed that as well